### PR TITLE
Update documentation to include Go SDK (backport #2377)

### PR DESCRIPTION
**Pull request by WaleedMortaja**
A backport of pull request #2377

### CODE REVIEW

**Comment by WaleedMortaja** [comment]
Why have I been considered the author of the commit while there is a 
`Signed-off-by: pama-ibm <pama@ibm.com>` line?

I have not changed any thing in the commit message except appending `(cherry picked from commit https://github.com/hyperledger/fabric/commit/c03222079a90f5ff9dc949bfa45a7047e9bcc33a)` line (The same as I had done for other accepted commits).

Is there any format error or something in the commit message?

File: docs/source/dev-setup/devenv.rst
Comment (denyeart) [inline docs/source/dev-setup/devenv.rst]: Keep as main

**Comment by denyeart** [comment]
@ryjones Do you know if the person cherry picking a fix to another branch needs to also sign the commit? Or is the original signature sufficient?

Maybe the DCO check only checks the very last line, and the cherry pick message interferes with the check?

For example in this case the commit says:

```
Signed-off-by: pama-ibm <pama@ibm.com>
(cherry picked from commit https://github.com/hyperledger/fabric/commit/c03222079a90f5ff9dc949bfa45a7047e9bcc33a)
```

**Comment by ryjones** [comment]
I'd move the cherry-pick decoration above the SoB line, @denyeart @WaleedMortaja

**Comment by WaleedMortaja** [comment]
@denyeart @ryjones I had actually been trying to do backporting the same way the bot `Mergifyio` does it.

I do not think the line of cherry-pick is interfereing with the check, because both the bot and I had previous commits with the same message format (i.e., a cherry-pick line being the last line. See [this](https://github.com/hyperledger/fabric/pull/3400/commits) for an example) and they were accepted.

I do not think it would be the cause of the problem but It may help to tell that I think I had used the command `git cherry-pick <sha>` in previous (accepted) commits. While I had `git cherry-pick -n <sha>` followed by `git commit` in this commit (So I would be able to change the message instead of committing and then amending).

In the commit in the example, the author of backported commit was show to be the same author of the original commit, and his sign is the only sign in the commit message. However, in this commit, the original author was dropped. I do not know why yet!

I will look into this and update you.

File: docs/source/dev-setup/devenv.rst
Comment (WaleedMortaja) [reply docs/source/dev-setup/devenv.rst]: It was backported with reset of the code. Thank you for notifying me. I will edit it .. after searching the DCO check issue.

**Comment by ryjones** [comment]
@WaleedMortaja if you enable "maintainers can edit PR" I will fix it up

**Comment by WaleedMortaja** [comment]
It is already enabled

**Comment by WaleedMortaja** [comment]
@ryjones Thank you. The DCO worked. What do yo think was the problem?

\+ Can you apply [the above requested edit](https://github.com/hyperledger/fabric/pull/3401#discussion_r869474720) before merging?

**Comment by ryjones** [comment]
@denyeart I think I've fixed it

**Comment by WaleedMortaja** [comment]
It should be `main` not `master`

**Comment by ryjones** [comment]
@WaleedMortaja done

**Comment by WaleedMortaja** [comment]
According to GitHub note: `WaleedMortaja authored and ryjones committed`. So I think `pama-ibm <pama@ibm.com>` is not taking any credit even though being the original author?! (By the way, I am not the author nor co-author. I have just cherry-picked a commit that I had found missing in release-2.2 branch)

**Review by denyeart — APPROVED** [review]

**Pull request merged**
